test/metabase/upload_test.clj

```diff
@@ -1511,6 +1511,42 @@
                    (rows-for-table table))))
           (io/delete-file file))))))
 
+(deftest csv-append-snowplow-test
+  ;; Just test with h2 because snowplow should be independent of the driver
+  (mt/test-driver :h2
+    (snowplow-test/with-fake-snowplow-collector
+
+     (with-upload-table! [table (create-upload-table!)]
```

Comment:
Can you leave some `testing` blocks here? One for the success case, and one for the failure case

Reply:
Good idea. I'll add for both creates and appends.